app/models/merchant.rb

```diff
@@ -4,4 +4,14 @@ class Merchant < ApplicationRecord
   has_many :transactions, through: :invoices
 
   validates :name, presence: true
+
+  def total_revenue
+    invoices.select(&:successful?).reduce(0) do |total, invoice|
+      total += invoice.total
+    end
+  end
+
+  def self.with_invoices
```

Comment:
@s-espinosa Here are the methods that are doing the calculations as well as doing some querying. Is this sufficient or this too much Ruby for your taste?